devtools/aesh/src/main/java/io/quarkus/cli/commands/AddExtensionCommand.java

```diff
@@ -1,73 +1,74 @@
 package io.quarkus.cli.commands;
 
+import java.io.IOException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 import org.aesh.command.Command;
 import org.aesh.command.CommandDefinition;
 import org.aesh.command.CommandException;
 import org.aesh.command.CommandResult;
 import org.aesh.command.invocation.CommandInvocation;
-import org.aesh.command.option.Argument;
 import org.aesh.command.option.Option;
+import org.aesh.command.option.OptionList;
 import org.aesh.io.Resource;
+import org.aesh.selector.MultiSelect;
 
 import io.quarkus.dependencies.Extension;
 import io.quarkus.devtools.project.QuarkusProject;
-import io.quarkus.platform.descriptor.QuarkusPlatformDescriptor;
 import io.quarkus.platform.tools.config.QuarkusPlatformConfig;
 
-/**
- * @author <a href="mailto:stalep@gmail.com">Ståle Pedersen</a>
- */
-@CommandDefinition(name = "add-extension", description = "Adds extensions to a project")
+@CommandDefinition(name = AddExtensions.NAME, description = "Adds extensions to a project")
 public class AddExtensionCommand implements Command<CommandInvocation> {
 
-    @Option(shortName = 'h', hasValue = false, overrideRequired = true)
-    private boolean help;
+    @OptionList(shortName = 'e', //selector = SelectorType.SELECTIONS,
+            completer = ExtensionCompleter.class, description = "Name of the extension that will be added to the project")
+    private Set<String> extensions;
 
-    @Option(shortName = 'e', required = true, description = "Name of the extension that will be added to the project")
-    private String extension;
-
-    @Argument(description = "path to the project", required = true)
+    @Option(shortName = 'p', description = "Path to the project, if not set it will use the current working directory")
     private Resource path;
 
     @Override
-    public CommandResult execute(CommandInvocation commandInvocation) throws CommandException, InterruptedException {
-        if (help) {
-            commandInvocation.println(commandInvocation.getHelpInfo("quarkus add-extension"));
-            return CommandResult.SUCCESS;
-        } else {
+    public CommandResult execute(CommandInvocation invocation) throws CommandException, InterruptedException {
+
+        try {
+            Path projectDirectory = path != null ? Paths.get(path.getAbsolutePath())
+                    : Paths.get(System.getProperty("user.dir"));
+
+            QuarkusProject quarkusProject = QuarkusProject.resolveExistingProject(projectDirectory,
+                    QuarkusPlatformConfig.getGlobalDefault().getPlatformDescriptor());
```

Comment:
I guess, it's fine to keep as you are doing a refactoring here but this getGlobalDefault() should not be used. We should get rid of it ASAP. You may be the only one using it now.

Reply:
I see, what is the preferred way of doing it?
There are quite a few changes since I last looked at this so I just used what I thought worked :)

Reply:
We are missing a convenient abstraction for that, which is going to complicate it for you and any non-maven/gradle tool. For Maven projects we do that in BuildFileMojoBase (look how we initialize platformDescr). In Gradle that would be QuarkusPlatformTask.platformDescriptor().

Reply:
I see, yes it is a bit difficult on the CLI side since the project can be either Maven/Gradle. Some moons ago I used BuildFile/BuildTool to try to figure out what it was. If we could have a fairly simple way to do that it would be helpful for the CLI.